pkg/nodediscovery/nodediscovery.go

```diff
@@ -454,7 +454,11 @@ func (n *NodeDiscovery) mutateNodeResource(nodeResource *ciliumv2.CiliumNode) er
 
 	nodeResource.ObjectMeta.Labels = k8sNodeParsed.Labels
 
-	localCN := n.localNode.ToCiliumNode()
+	// This is for syncing relevant node annotations from the k8s node to the
+	// CiliumNode object, ToCiliumNode will only copy used annotations.
+	ln := n.localNode.DeepCopy()
+	ln.Annotations = k8sNodeParsed.Annotations
+	localCN := ln.ToCiliumNode()
```

Comment:
AFAIU we are creating a temporary `localNode` and setting its annotations to `k8sNodeParsed.Annotations` so that `ToCiliumNode` will "extract" the used ones (WireguardPubKey and BGP virtual router) in`localCN`.
What about adding an exported method to `types.Node` to extract the used annotations and use it both here (on `k8sNodeParsed`) and inside `*Node.ToCiliumNode`?

Reply:
Good idea, did that!